src/NUnitEngine/nunit.engine/Services/TestAgency.cs

```diff
@@ -161,10 +161,9 @@ private Guid LaunchAgentProcess(TestPackage package)
 
             string agentExePath = GetTestAgentExePath(useX86Agent);
 
-            if (agentExePath == null)
-                throw new ArgumentException(
-                    string.Format("NUnit components for version {0} of the CLR are not installed",
-                    targetRuntime.ClrVersion.ToString()), "targetRuntime");
+            if (!File.Exists(agentExePath))
+                throw new FileNotFoundException(
+                    $"{Path.GetFileName(agentExePath)} for version {targetRuntime.ClrVersion.ToString()} of the CLR could not be found.", agentExePath);
```

Comment:
Ahh, thanks. I did wonder - I wondered if this was future proofed, as we've talked about having different agents to run on different platforms.

I've removed it - we can add it back when we need it. 😄